clcache.py

```diff
@@ -1371,24 +1379,24 @@ def processCacheHit(cache, objectFile, cachekey):
         return 0, cachedArtifacts.stdout, cachedArtifacts.stderr, False
 
 
-def createManifest(manifestHash, includePaths):
-    baseDir = normalizeBaseDir(os.environ.get('CLCACHE_BASEDIR'))
-
-    includes = {path:getFileHash(path) for path in includePaths}
-    includesContentHash = ManifestRepository.getIncludesContentHashForFiles(includes)
+def createManifestEntry(manifestHash, includePaths):
+    includesWithHash = [[path, getFileHash(path)] for path in includePaths]
+    includesContentHash = ManifestRepository.getIncludesContentHashForHashes(
+        [hash for include, hash in includesWithHash])
```

Comment:
If you keep `includesWithHash` (aka `includes` in the old version) a dictionary, you can simplify the `includesContentHash` definition (as well as `safeIncludes` further down) a bit by using `keys()` and `values()`, i.e. you don't need to unpack both values and then ignore one of them.

Not sure this actually improves anything, but a dictionary seems like a fairly natural data structure in this case to me, doesn't it?